presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/PlanMatchPattern.java

```diff
@@ -59,6 +59,11 @@ public static PlanMatchPattern anyTree(PlanMatchPattern... sources)
         return any(sources).matchToAnyNodeTree();
     }
 
+    public static PlanMatchPattern anyNot(Class<? extends PlanNode> excludeNodeClass, PlanMatchPattern... sources)
```

Comment:
This seems overly specific. Couldn't it be done by introducing a `not()` matcher and then composing `any`, `not` and `node`?